From alan@linux.intel.com  Mon Dec  7 16:59:56 2009
From: Alan Cox <alan@linux.intel.com>
Date: Mon, 30 Nov 2009 13:17:19 +0000
Subject: tty: n_tty: For some I/O patterns n_tty will lose bytes
To: greg@kroah.com
Message-ID: <20091130131719.26701.63040.stgit@localhost.localdomain>


Fix the obvious disagreement between the receive path and the receive room
logic.

Signed-off-by: Alan Cox <alan@linux.intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---

 drivers/char/n_tty.c |    3 +++
 1 file changed, 3 insertions(+)


--- a/drivers/char/n_tty.c
+++ b/drivers/char/n_tty.c
@@ -95,6 +95,9 @@ static void n_tty_set_room(struct tty_st
 	/* tty->read_cnt is not read locked ? */
 	int	left = N_TTY_BUF_SIZE - tty->read_cnt - 1;
 
+	if (tty->real_raw)
+	        left = min(left, N_TTY_BUF_SIZE - tty->read_head - 1);
+
 	/*
 	 * If we are doing input canonicalization, and there are no
 	 * pending newlines, let characters through without limit, so
